### DISCUSSION TITLE
fetching data from graphql api within dynamic route is returning 404

### DISCUSSION THREAD

**Discussion by nebulahh** [vercel/next.js#49381] [Help] [answered]
### Summary

i am fetching data from a graphql api and i want to navigate from a master page to a detail page using the Link tag to get the detail about a post. my initial method was to use getStaticPaths and pass the postId to getStaticProps and this worked for a while but recently the details page has been returning 404.

this is the link in the master page
```js
  <Link href={`/${id}`} passHref>
    <div className='text-white text-base'>
      {text}
   </div>
 </Link>
```

this is the detail page 

```ts
export default function Status({ tweet }) {
  return (
    <h1>{tweet.singleTweet.text}</h1>
)
}

export async function getStaticPaths() {
  const client = new ApolloClient({
  uri: 'http://localhost:4000/graphql',
  cache: new InMemoryCache(),
});

  const { data } = await client.query({
    query: gql`
    query {
    feed {
      text
      id
    }
  }`
  })

  const paths = data.feed.map((tweet: any) => {
    return {
      params: {
        postId: `${tweet.id}`
      }
    }
  })

  return {
    paths,
    fallback: false,
  }
}

export async function getStaticProps(context: any) {  
  const { params } = context

  const client = new ApolloClient({
    uri: 'http://localhost:4000/graphql',
    cache: new InMemoryCache(),
  });

  const { data } = await client.query({
    query: gql`
    query singleTweet {
      singleTweet(id: ${params.postId}) {
        text
        id
        createdAt
        postedBy{
          name
       }
       votes {
       id
      }
      comments {
        id
        body
      }
     createdAt
    }
  }`
  })

  return {
    props: {
      tweet: data
    }
  }
}

```

i saw this [sample](https://www.apollographql.com/blog/apollo-client/how-to-use-apollo-client-with-next-js-13/) from apolloclient, but it does not show how to handle dynamic parameters

**Comment by icyJoseph** [comment, marked as answer]
Well, I'd start by adding some logs, make sure your assumptions are correct. For example, what value does `params.postId` have?

Also, since `getStaticPaths` has fallback set to false, only the paths your return there, are allowed. Everything else will yield 404. 

So log the contents of `paths` in `getStaticPaths`, can your reach those?
